File: app/src/main/java/srct/whatsopen/model/SpecialSchedule.java
package srct.whatsopen.model;

import com.google.gson.annotations.SerializedName;

import io.realm.RealmList;
import io.realm.RealmObject;

public class SpecialSchedule extends RealmObject implements Schedule {

    public SpecialSchedule(RealmList<OpenTimes> openTimesList, String validStart, String validEnd) {
        mOpenTimesList = openTimesList;
        this.validStart = validStart;
        this.validEnd = validEnd;
    }

    public SpecialSchedule() {
    }

    @SerializedName("open_times")
    private RealmList<OpenTimes> mOpenTimesList;

    @SerializedName("valid_start")
    private String validStart;

    @SerializedName("valid_end")
    private String validEnd;

    public String getValidStart() {
        return validStart;
    }

    public void setValidStart(String validStart) {
        this.validStart = validStart;
    }

    public String getValidEnd() {
        return validEnd;
    }

    public void setValidEnd(String validEnd) {
        this.validEnd = validEnd;
    }

    public RealmList<OpenTimes> getOpenTimesList() {
        return mOpenTimesList;
    }

    public void setOpenTimesList(RealmList<OpenTimes> openTimesList) {
        mOpenTimesList = openTimesList;
    }
}

File: app/src/main/java/srct/whatsopen/model/SpecialSchedule.kt
package srct.whatsopen.model

import com.google.gson.annotations.SerializedName

import io.realm.RealmList
import io.realm.RealmObject

open class SpecialSchedule : RealmObject, Schedule {

    @SerializedName("open_times")
    private var mOpenTimesList: RealmList<OpenTimes>? = null

    @SerializedName("valid_start")
    var validStart: String? = null

    @SerializedName("valid_end")
    private var validEnd: String? = null

    constructor(openTimesList: RealmList<OpenTimes>, validStart: String, validEnd: String) {
        mOpenTimesList = openTimesList
        this.validStart = validStart
        this.validEnd = validEnd
    }

    constructor() {}

    override fun getValidEnd(): String? {
        return validEnd
    }

    fun setValidEnd(validEnd: String) {
        this.validEnd = validEnd
    }

    override fun getOpenTimesList(): RealmList<OpenTimes>? {
        return mOpenTimesList
    }

    fun setOpenTimesList(openTimesList: RealmList<OpenTimes>) {
        mOpenTimesList = openTimesList
    }
}
